code/game/objects/structures/extinguisher.dm

```diff
@@ -40,8 +57,11 @@
 			user.drop_item(O)
```

Comment:
not safe against `NO_DROP`

Reply:
I have added sanity check for no drop case. Fixed.